install/kubernetes/cilium/templates/hubble-ui/deployment.yaml

```
@@ -29,9 +29,8 @@ spec:
     spec:
       {{- if .Values.hubble.ui.securityContext.enabled }}
```

Since we are deprecating `.Values.hubble.ui.securityContext.enabled`, we need to think about a clear upgrade path for when it will be removed.

I'm wondering what will be the equivalent of `--set hubble.ui.securityContext.enabled=false`. What comes to mind is `--set hubble.ui.securityContext=null` which doesn't currently work:

```
Error: template: cilium/templates/hubble-ui/deployment.yaml:30:20: executing "cilium/templates/hubble-ui/deployment.yaml" at <.Values.hubble.ui.securityContext.enabled>: nil pointer evaluating interface {}.enabled
```

So I think we need to handle the `null` case in this PR for forward-compatibility (i.e. compatibility for when the `enabled` flag will be removed).

```suggestion
      {{- if and .Values.hubble.ui.securityContext .Values.hubble.ui.securityContext.enabled }}
```

Does that make sense? cc @sayboras

---

`and` will evaluate both parameters, so it will fail too. If we want to handle `null`, then `with` is a better choice, like this:

```yaml
      {{- with .Values.hubble.ui.securityContext }}
        {{- if .enabled }}
      securityContext:
        {{- $securityContext := omit . "enabled" }}
        {{- toYaml $securityContext | nindent 8 }}
        {{- end}}
      {{- end }}
```

---

> `and` will evaluate both parameters, so it will fail too. 

This is not my understanding of how `and` works (emphasis mine):

> Returns the boolean AND of its arguments by returning the first empty argument or the last argument. _That is, "and x y" behaves as "if x then y else x."_
> Evaluation proceeds through the arguments left to right and returns when the result is determined.
> — https://pkg.go.dev/text/template#hdr-Functions

I tested the suggestion beforehand and it worked for me, do you get an error?

---

It failed on my machine because my helm is built using go 1.17, that behavior changed in 1.18 https://tip.golang.org/doc/go1.18

> The and function no longer always evaluates all arguments; it stops evaluating arguments after the first argument that evaluates to false. Similarly, the or function now stops evaluating arguments after the first argument that evaluates to true. This makes a difference if any of the arguments is a function call.

---

btw, new change pushed, not sure if the upgrade note is in the correct position and appropriate, can you take a look, thanks!

---

TIL about this change between Go v1.17 and v1.18! Will test and check the upgrade note thank you.